homeassistant/components/generic/__init__.py

```diff
@@ -1,6 +1,21 @@
 """The generic component."""
 
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
 
 DOMAIN = "generic"
 PLATFORMS = [Platform.CAMERA]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up generic IP camera from a config entry."""
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
```

Comment:
I think we also need a config entry change listener to reload when it changes the title or options

Reply:
Example #68147

Reply:
Hmm, I am not sure how to do that. Do you have any examples?

Reply:
See comment above

Reply:
Ok, but not sure why this is needed.  Actually I've not implicitly linked the title and entity name.  Entity name defaults to the title until the user enters something or if one was defined in the imported yaml - that seems to be how other integrations behave.

But it could well be that I've not fully understood what you were looking for.

Reply:
If they change the title the entity name won't update until its reloaded

Reply:
Is that a normal pattern, e.g. for renaming an integration title to result from in automatic renames of the devices or entities beneath it?

Personally I think it makes sense and I do think it is confusing to be able to assign name and title differently.  But I see the same behaviour in the esphome integration, so I thought that was normal.

Wouldn't mapping title->name also mean we'd need to map name->title?

Hopefully there is a doc or architecture decision clarifying this but I don't know where.

Reply:
In general we are moving away from collecting a name when it is a 1:1 relationship and using the title.  It is a more recent development so I don't think there is any documentation on it.

Reply:
Done.